lib/jekyll-readme-index/generator.rb

```diff
@@ -1,7 +1,7 @@
 module JekyllReadmeIndex
   class Generator < Jekyll::Generator
     README_REGEX = %r!^/readme(\.[^.]+)?$!i
-    INDEX_REGEX = %r!^/($|index\.(html?|xhtml|xml)$)!i
```

Comment:
What's the benefit of moving the constant to the method? I find these types of regexes often get tweaked over time (as we recently did with `xml` and `xhtml` and it's easy for non-Ruby or non-Jekyll contributors to do so when it's distinct from the logic.